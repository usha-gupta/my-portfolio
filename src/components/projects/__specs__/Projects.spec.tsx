import { act, fireEvent, render, waitFor } from "@testing-library/react";
import Projects from "../Projects";

describe("Projects", () => {
  it("verify snapshot for projects", () => {
    const { asFragment } = render(<Projects />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("project detail modal should open on click of project card", () => {
    const { getAllByTestId, getByTestId } = render(<Projects />);
    const projectCard = getAllByTestId("project-card");
    act(() => {
      fireEvent.click(projectCard[0]);
    });

    expect(getByTestId("modal-project-title").textContent).toEqual(
      getAllByTestId("project-title")[0].textContent
    );
  });

  it("project detail modal should close on click of close icon", async () => {
    const { getAllByTestId, queryByTestId, getByRole } = render(<Projects />);
    const projectCard = getAllByTestId("project-card");
    act(() => {
      fireEvent.click(projectCard[0]);
    });

    await waitFor(() => {
      expect(queryByTestId("modal-project-title")).not.toBeNull();
    });

    act(() => {
      fireEvent.click(getByRole("button", { name: "close" }));
    });

    await waitFor(() => {
      expect(queryByTestId("modal-project-title")).toBeNull();
    });
  });
});
