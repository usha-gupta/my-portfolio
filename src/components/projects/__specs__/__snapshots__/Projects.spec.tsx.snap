// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects verify snapshot for projects 1`] = `
<DocumentFragment>
  .c0 {
  height: 60px;
  text-overflow: ellipsis;
  overflow: hidden;
  display: -webkit-box !important;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  white-space: normal;
}

<div
    class="MuiBox-root css-eeo3ga"
    id="projects-section"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-1oe6gg4-MuiTypography-root"
    >
      Projects
    </h3>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-6s1xft-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-hhotpj-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
          data-testid="project-card"
        >
          <picture>
            <source
              srcset="test-file-stub"
              type="image/webp"
            />
            <source
              srcset="test-file-stub"
              type="image/jpeg"
            />
            <img
              alt="Paisa Vasool screenshot"
              src="test-file-stub"
              style="display: block; width: 100%; height: 140px; object-fit: cover;"
            />
          </picture>
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-14szp5y-MuiTypography-root"
              data-testid="project-title"
            >
              Paisa Vasool
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 c0 css-b3lvtk-MuiTypography-root"
            >
              It's a expense management and tracker app which also let you split the expenses among your friends and groups.
            </p>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Learn More
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
